examples/microphone_nextjs/pages/index.tsx

```diff
@@ -0,0 +1,270 @@
+        },
+      });
+
+    }).catch(err => setSessionState("blocked"))
+  };
+
+  // Stop the transcription on click to end the recording
+  const stopTranscription = async () => {
+    await audioRecorder.stopRecording();
+    await rtSessionRef.current.stop();
+  };
+
+  return (
+    <div>
+      <div className="flex-row">
+        <p>Select Microphone</p>
+        {sessionState === "blocked" && (
+          <p className="warning-text">Microphone permission is blocked</p>
+        )}
+      </div>
+      <MicSelect
+        onClick={() => {
+          setRequestDevices(true);
+        }}
+        value={audioDeviceId}
+        options={devices.map((item) => {
+          return { value: item.deviceId, label: item.label };
+        })}
+        onChange={(e) => setAudioDeviceId(e.target.value)}
+      />
+      <TranscriptionButton
+        sessionState={sessionState}
+        stopTranscription={stopTranscription}
+        startTranscription={startTranscription}
+      />
+      {sessionState === "error" && (
+        <p className="warning-text">Session encountered an error</p>
+      )}
+      {["starting", "running", "configure", "blocked"].includes(
+        sessionState
+      ) && <p>State: {sessionState}</p>}
+      <p>
+        {transcription.map(
+          (item, index) =>
+            (index && ![".", ","].includes(item.alternatives[0].content)
+              ? " "
+              : "") + item.alternatives[0].content
+        )}
+      </p>
+    </div>
+  );
+}
+
+// getServerSideProps - used to perform server side preparation
+// In this case, the long-lived API key is provided to the server and used to fetch a short-lived JWT
+// The short-lived JWT is then given to the client to connect to Speechmatics' service
+// This ensures the security of long-lived tokens and reduces the scope for abuse from end users
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const jwt = await getJwt();
+  if (jwt === undefined) throw new Error("JWT undefined");
+  return {
+    props: { jwt },
+  };
+};
+
+// ButtonInfoBar - component for stopping/starting session
+
+type TranscriptionButtonProps = {
+  startTranscription: () => void;
+  stopTranscription: () => void;
+  sessionState: SessionState;
+};
+
+function TranscriptionButton({
+  startTranscription,
+  stopTranscription,
+  sessionState,
+}: TranscriptionButtonProps) {
+  return (
+    <div className="bottom-button-status">
+      {["configure", "stopped", "starting", "error", "blocked"].includes(
+        sessionState
+      ) && (
+        <button
+          type="button"
+          className="bottom-button start-button"
+          disabled={["starting", "blocked"].includes(sessionState)}
+          onClick={async () => {
+            startTranscription();
+          }}
+        >
+          <CircleIcon style={{ marginRight: "0.25em", marginTop: "1px" }} />
+          Start Transcribing
+        </button>
+      )}
+
+      {sessionState === "running" && (
```

Comment:
At the moment, when a user hits transcribe, they have to wait for transcription to start in order to stop it.

It would be good if the user could abort early, such as if they pressed the button by mistake. Perhaps this just involves showing a stop button when sessionState is 'starting'?

Reply:
It depends on what stage we would cancel the process at. Once we've started to connect to the websocket, we need to allow the websocket connection to disconnect gracefully, so there's always going to be a slight delay in connecting/disconnecting for those scenarios. That being said we can leave the button active throughout, and then trigger a shutdown of the websocket on click